packages/fuels/src/cli/utils/checkForAndDisplayUpdates.ts

```diff
@@ -0,0 +1,42 @@
+import { versions, gt, eq } from '@fuel-ts/versions';
+
+import { warn, log } from './logger';
+
+export const getLatestFuelsVersion = async () => {
+  const response = await fetch('https://registry.npmjs.org/fuels/latest');
+  const data = await response.json();
+  return data.version as string;
+};
+
+export const checkForAndDisplayUpdates = async () => {
+  try {
+    const { FUELS: userFuelsVersion } = versions;
+
+    const latestFuelsVersion = await Promise.race<string | undefined>([
+      new Promise((resolve) => {
+        setTimeout(resolve, 3000);
+      }),
+      getLatestFuelsVersion(),
+    ]);
+
+    if (!latestFuelsVersion) {
+      return;
+    }
+
+    const isFuelsVersionOutdated = gt(latestFuelsVersion, userFuelsVersion);
+    const isFuelsVersionUpToDate = eq(latestFuelsVersion, userFuelsVersion);
+
+    if (isFuelsVersionOutdated) {
+      warn(
+        `\n⚠️ There is a newer version of fuels available: ${latestFuelsVersion}. Your version is: ${userFuelsVersion}\n`
+      );
+      return;
+    }
+
+    if (isFuelsVersionUpToDate) {
+      log(`\n✅ Your fuels version is up to date: ${userFuelsVersion}\n`);
+    }
+  } catch {
+    /* Fail gracefully */
```

Comment:
I agree that we should fail gracefully, but I do wonder how we would ever know if this is failing for consumers if we don't log something. Perhaps we should at least acknowledge that we were unable to fetch the latest fuel version.

```suggestion
log(`\n currently unable to fetch latest fuels version. Skipping`)
```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3108/commits/5df5b7868361f4e8d5b3f6f44d7f5052650988ec